### ISSUE TITLE
obspy.read_inventory doesn't work with stationXML

### ISSUE THREAD

**Issue by SCEDC** [obspy/obspy#1060]
We enjoyed the Obspy presentation given today by Lion.  
I tried using the obspy.read_inventory on one of our stationXML files and it failed.
Here is the file: http://files.anss-sis.scsn.org/production/FDSNstationXML/CI/CIBAR.xml
Here is the error:
![screen shot 2015-05-27 at 9 07 47 am](https://cloud.githubusercontent.com/assets/1987779/7841226/fcb685e6-044f-11e5-8e51-d0c4d84f89c0.png)

**Comment by krischer** [comment]
Strange error. Can you upload the file again? It somehow did not work. Thanks!

**Comment by SCEDC** [comment]
I'm sorry -- the file is located at:
http://files.anss-sis.scsn.org/production/FDSNstationXML/CI/CI_BAR.xml

**Comment by krischer** [comment]
The error is because your file does not have a default namespace. ObsPy assumes (which is actually wrong), that the default namespace is the StationXML namespace. I'll fix it.

This is the first StationXML file I see where this happen and we definitely did parse a fair share of files out there.

Can I include parts of the file to ObsPy for testing purposes?

**Comment by krischer** [comment]
Changing line 96 in `//anaconda/.../obspy/station/stationxml.py` from

```python
namespace = root.nsmap[None]
```

to

```python
namespace = "http://www.fdsn.org/xml/station/1"
```

will patch your local installation.

**Comment by krischer** [comment]
Actually nevermind...its simple enough to create my own completely independent file. Thanks for the report!

**Comment by SCEDC** [comment]
Yes you can use this file.  This file was produced by our Station Information System which is a seismic station metadata repository for a number ANSS seismic networks so if you find any errors, please let us know too as would affect a large number of stationXML files out there.  (sis-help@gps.caltech.edu)

**Comment by krischer** [comment]
The file itself is fine.

If you have the power to change these files it might be worth it to set the default namespace to the StationXML namespace. Other software might make the same assumption as ObsPy did and thus it will not work. As I said: we never encountered another file with this so far.

**Comment by SCEDC** [comment]
Ok - Let me talk with our developer about getting it changed.  I also had problems plotting the response curve:

![screen shot 2015-05-27 at 10 01 47 am](https://cloud.githubusercontent.com/assets/1987779/7842371/8e188bfe-0457-11e5-8bdb-7751824758d6.png)

**Comment by SCEDC** [comment]
re: namespace - are you referring to this string in the document:
xmlns="http://www.fdsn.org/xml/station/1"?
I could be mistaken but that seems to be the string returned by the IRIS webservice?

http://service.iris.edu/fdsnws/station/1/query?network=TA&station=A*&channel=BH?,-BHZ&level=channel&format=xml

**Comment by krischer** [comment]
The string is fine; your StationXML is valid after all.

IRIS uses `xmlns="http://www.fdsn.org/xml/station/1"` which puts all elements that have no namespace prefix under the StationXML namespace. In your file you use `xmlns:fsx="http://www.fdsn.org/xml/station/1"` meaning all StationXML elements have to be prefixed with `fsx`. Now this is completely valid and was just an error from our side. I just thought other programs might do the same so if its a simple change it might be worth it.

Regarding the response plot: The plot function estimates the nyquist frequency from the sampling rate of the channel. In your file its set to zero in a couple of cases:

```bash
$ grep SampleRate CI_BAR.xml | grep "0\.0000"

        <fsx:SampleRate>0.000000000000E+00</fsx:SampleRate>
              <fsx:InputSampleRate>0.000000000000E+00</fsx:InputSampleRate>
        <fsx:SampleRate>0.000000000000E+00</fsx:SampleRate>
              <fsx:InputSampleRate>0.000000000000E+00</fsx:InputSampleRate>
        <fsx:SampleRate>0.000000000000E+00</fsx:SampleRate>
              <fsx:InputSampleRate>0.000000000000E+00</fsx:InputSampleRate>
        <fsx:SampleRate>0.000000000000E+00</fsx:SampleRate>
              <fsx:InputSampleRate>0.000000000000E+00</fsx:InputSampleRate>
        <fsx:SampleRate>0.000000000000E+00</fsx:SampleRate>
              <fsx:InputSampleRate>0.000000000000E+00</fsx:InputSampleRate>
        <fsx:SampleRate>0.000000000000E+00</fsx:SampleRate>
              <fsx:InputSampleRate>0.000000000000E+00</fsx:InputSampleRate>
```

This happens in your `ACE` channels - I am not sure if that is allowed honestly. Maybe our plot_reponse() function should be a bit smarter in theses cases.

Filtering it a bit does the trick:

```python
inv.select(channel="BH?").plot_response(0.001)
```

**Comment by megies** [comment]
Reopening for the plotting issue..

**Comment by QuLogic** [comment]
> I am not sure if that is allowed honestly

The StationXML schema page says:
> > When definitions and usage are underdefined the SEED manual should be referred to for clarification.

And page 68 of the SEED manual states:
> > Set this rate to zero for channels not sampled at regular intervals (such as console logs or alarms).

So I'd say yes, it's allowed.

**Comment by QuLogic** [comment]
So two things: 1) Why include a Decimation as part of the ResponseStage for something with no real sample rate? 2) if the Decimation is removed, then ObsPy needs some modification to handle plotting of a response that consists solely of a StageGain.

**Comment by QuLogic** [comment]
Also note, the schema does not claim the SampleRate is required on the Channel.

My first thought of falling back to the StageGain->Frequency will not work because one of your ACE channels is 1 Hz while the other is 0 Hz.

**Comment by megies** [comment]
Closed again by #1115.
